r/tests/testthat/test-ipc.R

```diff
@@ -0,0 +1,216 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+test_that("read_nanoarrow() works for raw vectors", {
```

Comment:
Would it make sense to add tests here for non-happy-path behavior? Malformed raw vectors, wrong file type, etc? If yes, I suppose this would make sense to match in the python bindigs.

For example, trying to read an Arrow IPC file results in,

```r
> read_nanoarrow("~/Data/starwars.arrow")
<nanoarrow_array_stream[<error calling get_schema()]>
 $ get_schema:function ()
 $ get_next  :function (schema = x$get_schema(), validate = TRUE)
 $ release   :function ()
```